### PR TITLE
Update repo and packaging set-up

### PR DESCRIPTION
**Pull request by jacksonllee**
This PR updates the repo and packaging set-up:

* Bump `wikipron` package version to 1.3.0.
* Convert `setup.py` into `pyproject.toml`.
* Add support for Python 3.11.
* Drop support for Python 3.7 (its EOL is mid 2023; most major open-source projects like numpy, pandas, and such have already dropped Python 3.7).
* Add the twine checks for the Python source distribution and wheel on CI.
* Turn on tests for Windows on CI.

---

- [x] Updated `Unreleased` in `CHANGELOG.md` to reflect the changes in code or data.

### CODE REVIEW

File: data/scrape/scrape.py
Comment (jacksonllee) [inline data/scrape/scrape.py:52]: All code/typing changes are due to the newer mypy version being more stringent.

**Comment by jacksonllee** [comment]
After this PR is merged, I'll make a new `wikipron` package release (both PyPI and GitHub).

**Review by kylebgorman — APPROVED** [review]
LGTM. Do you have approval to submit this or should I?

**Pull request merged**
